### PR TITLE
feat: [CDS-87802]: Preference framework backend integration and added toggle button to enable/disable the new nav

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

Added a toggle button in user profile popup to switch between the old nav and new nav. 

#### Screenshots

Screenshot 2024-01-05 at 9 15 28 AM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by avinashmadhwani02** [comment]
trigger messagecheck

File: src/modules/10-common/navigation/MainNav/MainNav.tsx [hunk 1]
```diff
@@ -193,18 +191,28 @@ export default function L1Nav(): React.ReactElement {
           </IfPrivateAccess>
           <li className={css.navItem}>
             <IfPrivateAccess>
-              <Link
-                className={cx(css.navLink, css.userLink, css.hoverNavLink)}
-                activeClassName={css.active}
-                to={paths.toUser(params)}
+              <Popover
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/navigation/MainNav/MainNav.tsx:194]: Can we cover this under jest?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/framework/AppStore/AppStoreContext.tsx [hunk 2]
```diff
@@ -488,6 +492,18 @@ export function AppStoreProvider({ children }: PropsWithChildren<unknown>): Reac
     }))
   }, [accountId])
 
+  React.useEffect(() => {
+    if (Object.keys(state.featureFlags).length > 1) {
+      const isNewNavPreferenceEnabled = userInfo?.data?.userPreferences?.['enable_new_nav'] === 'true'
+      if (state.featureFlags['CDS_NAV_2_0'] || isNewNavPreferenceEnabled) {
```
Comment (mayankVermaHarness) [inline src/framework/AppStore/AppStoreContext.tsx:498]: Should this be `&&` instead of `||` ?

Comment (avinashmadhwani02) [reply src/framework/AppStore/AppStoreContext.tsx:498]: It should be ||

if the feature flag is true - new nav should be enabled always and toggle button will not appear.
if the feature flag is false - user can see new nav through preferences (toggle button)

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Comment by avinashmadhwani02** [comment]
@abhinavrastogi-harness I'll fix the PR comments in a separate PR. Ticket for the same - https://harness.atlassian.net/browse/CDS-88085

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/framework/AppStore/AppStoreContext.tsx [hunk 3]
```diff
@@ -488,6 +492,18 @@ export function AppStoreProvider({ children }: PropsWithChildren<unknown>): Reac
     }))
   }, [accountId])
 
+  React.useEffect(() => {
+    if (Object.keys(state.featureFlags).length > 1) {
+      const isNewNavPreferenceEnabled = userInfo?.data?.userPreferences?.['enable_new_nav'] === 'true'
```
[the remark this answers is not in the document]
Comment (mayankVermaHarness) [reply src/framework/AppStore/AppStoreContext.tsx:497]: +1

**Pull request merged**